File: test-suite/src/test/java/io/micronaut/docs/inject/retainable/MinLength.java
package io.micronaut.docs.inject.retainable;

//tag::imports[]
import io.micronaut.context.annotation.AliasFor;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
//end::imports[]

//tag::class[]
@Limit(min = 5) // <1>
@Retention(RetentionPolicy.RUNTIME)
public @interface MinLength {

    @AliasFor(annotation = Limit.class, member = "min", applyDefault = true) // <2>
    int value() default 5;
}
//end::class[]

File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/inject/retainable/MinLength.kt
package io.micronaut.docs.inject.retainable

//tag::imports[]
import io.micronaut.context.annotation.AliasFor
//end::imports[]

//tag::class[]
@Limit(min = 5) // <1>
@Retention(AnnotationRetention.RUNTIME)
annotation class MinLength(
    @get:AliasFor(annotation = Limit::class, member = "min", applyDefault = true) // <2>
    val value: Int = 5
)
//end::class[]
